### PR TITLE
Linting Adjustments

### PR DESCRIPTION
**Pull request by gblakeman**
The bump in dependencies in #115 introduced a few linting rule adjustments. I commented on the few tweaks I made in `.eslintrc.js`, but overall the changes are really minor. The bulk of this PR are docs tweaks to conform my docs to some updates in the jsdoc rules.

----

Includes #115

### CODE REVIEW

File: src/data/run-data-update.mjs [hunk 1]
```diff
@@ -191,6 +190,7 @@ const pullAPIData = async () => {
   const callAndStoreData = async (currentTable) => {
     // ----- set up our faux `setTimeout` with async
     // can’t use `setTimeout` async, but can just force an async wait before moving on
+    // eslint-disable-next-line no-promise-executor-return
```
Comment (gblakeman) [inline src/data/run-data-update.mjs:193]: This is a new rule. I like it in general, but in this particular case, it seems worth deviating.

File: .eslintrc.js [hunk 2]
```diff
@@ -88,7 +88,14 @@ module.exports = {
       "ts": "never",
       "tsx": "never",
       "svelte": "never",
-    }]
+    }],
+    "default-param-last": 0,
```
Comment (gblakeman) [inline .eslintrc.js:92]: In totally new code, I like the idea of any params with default values going at the end of the function, but I worry we’d be opening up an error-prone can of worms to use this rule in our existing projects.

File: .eslintrc.js [hunk 3]
```diff
@@ -88,7 +88,14 @@ module.exports = {
...
+    "function-paren-newline": ["error", "consistent"],
```
Comment (gblakeman) [inline .eslintrc.js:93]: Setting this rule to `consistent` effectively keeps it at status quo with how we’ve been treating new lines around parentheses.

File: .eslintrc.js [hunk 4]
```diff
@@ -88,7 +88,14 @@ module.exports = {
...
+    "no-multiple-empty-lines": ["error", {
```
Comment (gblakeman) [inline .eslintrc.js:94]: Relaxing this rule is a bummer to me, but there [seems to be an error](https://github.com/sveltejs/eslint-plugin-svelte3/issues/41) in either `8.x` of `eslint` or in the `eslint-plugin-svelte3` package causing the `<script>` block in our Svelte files to count as an empty line. Hopefully we can go back to enforcing single blank lines in the future.

Comment (lynnerang) [reply .eslintrc.js:94]: yeahhh agreed, I'll take a mental note to keep an eye out and do this manually in the meantime

**Pull request merged**